internal/config/server/config.go

```diff
@@ -137,3 +139,16 @@ func setViperStructDefaults(v *viper.Viper, prefix string, s any) {
 		v.SetDefault(valueName, defaultValue)
 	}
 }
+
+func overrideViperStructDefaults(v *viper.Viper, prefix string, newDefaults string) {
+	overrides := map[string]any{}
+	if err := json.Unmarshal([]byte(newDefaults), &overrides); err != nil {
+		panic(fmt.Sprintf("Failed to parse overrides in %q: %v", prefix, err))
+	}
+
+	for key, value := range overrides {
+		// TODO: we don't do any fancy type checking here, so this could blow up later.
+		// I expect it will blow up at config-parse time, which should be earlier enough.
+		v.SetDefault(prefix+"."+key, value)
```

Comment:
Yeah, this can blow up if types don't match, but it will blow up while unmarshalling from Viper (when we`ReadConfigFromViper`), so this seems alright.

Reply:
Given the current use-cases, this seemed like the right balance between a large amount of type-matching code and deferring the error slightly later.